pkg/clustermanager/cluster_manager.go

```diff
@@ -34,29 +33,35 @@ import (
 )
 
 const (
-	maxRetries             = 30
-	backOffPeriod          = 5 * time.Second
-	machineBackoff         = 1 * time.Second
-	machinesMinWait        = 30 * time.Minute
-	moveCAPIWait           = 15 * time.Minute
-	clusterWaitStr         = "60m"
-	ctrlPlaneWaitStr       = "60m"
-	deploymentWaitStr      = "30m"
-	ctrlPlaneInProgressStr = "1m"
-	etcdInProgressStr      = "1m"
+	maxRetries                 = 30
+	DefaultBackOffPeriod       = 5 * time.Second
+	DefaultMachineBackoff      = 1 * time.Second
+	DefaultMachinesMinWait     = 30 * time.Minute
+	moveCAPIWait               = 15 * time.Minute
+	DefaultMaxWaitPerMachine   = 10 * time.Minute
+	clusterWaitStr             = "60m"
+	DefaultControlPlaneWaitStr = "60m"
+	deploymentWaitStr          = "30m"
+	controlPlaneInProgressStr  = "1m"
+	etcdInProgressStr          = "1m"
+	DefaultEtcdWaitStr         = "60m"
 )
 
+var DefaultPerMachineWaitStr = DefaultMaxWaitPerMachine.String()
+
 type ClusterManager struct {
 	*Upgrader
-	clusterClient      *retrierClient
-	writer             filewriter.FileWriter
-	networking         Networking
-	diagnosticsFactory diagnostics.DiagnosticBundleFactory
-	Retrier            *retrier.Retrier
-	machineMaxWait     time.Duration
-	machineBackoff     time.Duration
-	machinesMinWait    time.Duration
-	awsIamAuth         AwsIamAuth
+	clusterClient           *retrierClient
+	writer                  filewriter.FileWriter
+	networking              Networking
+	diagnosticsFactory      diagnostics.DiagnosticBundleFactory
+	Retrier                 *retrier.Retrier
+	machineMaxWait          time.Duration
+	machineBackoff          time.Duration
+	machinesMinWait         time.Duration
+	awsIamAuth              AwsIamAuth
+	controlPlaneWaitTimeout string
+	externalEtcdWaitTimeout string
```

Comment:
Could we store these two as `time.Duration` as well?

Reply:
I made them strings because they are all actually consumed by cluster manager as a string. The string form of these are what gets passed to kubectl wait so I left them as strings since that's what the library currently uses them as.

Reply:
Ahhh I see, yeah. However I would vote to keep them all as duration for api consistency. I think converting from duration to string is easy, right?

Reply:
Yes easy peasy I'll make them the same